airbyte-workers/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -327,4 +342,48 @@ public Long getTotalDestinationStateMessagesEmitted() {
     return totalDestinationEmittedStateMessages.get();
   }
 
+  @Override
+  public Long getMaxSecondsToReceiveSourceStateMessage() {
+    return maxSecondsToReceiveSourceStateMessage.get();
+  }
+
+  @Override
+  public Long getMeanSecondsToReceiveSourceStateMessage() {
+    return meanSecondsToReceiveSourceStateMessage.get();
+  }
+
+  private void updateMaxAndMeanSecondsToReceiveStateMessage(final DateTime stateMessageReceivedAt) {
+    final Long secondsSinceLastStateMessage = calculateSecondsSinceLastStateEmitted(stateMessageReceivedAt);
+    if (maxSecondsToReceiveSourceStateMessage.get() < secondsSinceLastStateMessage) {
+      maxSecondsToReceiveSourceStateMessage.set(secondsSinceLastStateMessage);
+    }
+
+    if (meanSecondsToReceiveSourceStateMessage.get() == 0) {
```

Comment:
I am wondering why are we using `AtomicLong`? Overall, it feels like we should be either using a Long or trying to leverage the atomic updates such as `compareAndExchange`.

Reply:
I was mostly just following the convention in the messagetracker for how we're tracking the other data